dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceFile.java

```diff
@@ -371,4 +399,16 @@ public static class VerificationInformation {
 
         // There is no dbupdatedate because it doesn't work with @Embeddable nor @ElementCollection
     }
+
+    public enum State {
+        COMPLETE,
+        MESSAGE,
+        STUB
+    }
+
+    public enum Reason {
```

Comment:
Open to considering the two enum approach. The weird part in my eyes is we would need constraints or a validator (?) to enforce illegal combinations. Like a complete + binary error for example.

What is the argument against one enum with more states?

Reply:
> What is the argument against one enum with more states?

Each enum contains a different type of information: `State` describes how the SourceFile is stored, and `Reason` describes   _why_ we stored the SourceFile that way.  When the values are separate, the code that uses them is simpler, and needs less modification when we add values, we don't need as many `ifs`.  The UI can look at `State` to figure out how to display the SourceFile, and other code can examine `Reason` to generate statistics, take remedial action, etc. 

What's the advantage to "one enum" (that encodes both values) besides eliminating the illegal combos?

It would be easy to add a constraint that made sure "reason" was null for the `COMPLETE` and `STUB` states.

Reply:
I'm on the fence on this one. Some thoughts:

Would you also need to add a constraint that if the `state` is `MESSAGE`, then `reason` has to be set? 

And if you add more states, you'd have to constrain those relationships too. Could it start to get a little complex?

On the other hand, if you have one enum, you could end up with a lot of values.

Digression on other possible states -- truncated, which you proposed. Not sure what else, maybe quarantined, lol -- we find or are informed about a file that might be have malware or PII?

Reply:
Both the "one" and "two" enum solutions seem to have their pros and cons.  How's about, in the interest of forward progress, we go with the "two enum" solution for now.  Maybe its pros will be more apparent when in use, and if we don't like it, we can easily change to the other later.   Agile!

Reply:
I should also note that I'm personally a proponent of the "one enum" solution with values that only encode the representation, with the "reason" being recorded as an English message, stored in the `content` field (for now).  Clean, simple, no partial 2d grid of enum values to have to keep consistent.

I separated the `Reason` field into a separate enum because a conversation with a key team member implied that they would insist upon it...

Reply:
Not wedded to the one enum approach, but thanks for explaining. 
I would prefer constraints or whatever in this PR to ensure  we don't forget and that we never have to deal with inconsistent data